### PR TITLE
Add TDD approach

### PR DESCRIPTION
**Pull request by NatiDeme**
# Changes in this PR:
- No linter errors
- Implemented solver class
- Implemented tests 
- Implemented methods

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,75 +1,41 @@
-![](https://img.shields.io/badge/Microverse-blueviolet)
-
-# Project Name
-
-> Description the project.
+# OOP school library
```
Comment (DammyShittu) [inline README.md]: - [x] Kindly rename the project title here

File: spec/factorial_spec.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  before(:each) do
+    @solver = Solver.new
+  end
+
+  it 'test factorial output' do
+    expect(@solver.factorial(5)).to eq(120)
+  end
+  it 'test factorial output' do
+    expect { @solver.factorial(-5) }.to raise_error
+  end
```
Comment (DammyShittu) [inline spec/factorial_spec.rb:13]: - [x] Include a test for a special scenario when the number is 0.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @NatiDeme @yohannesdagnachew,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mugishasam123 — APPROVED** [review]
Hi @NatiDeme  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- No linter errors 👍 
- Good commits ✔️ 
- Descriptive PR title ✔️ 

## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**